akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -30,6 +30,14 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   def isEmpty: Boolean
 
   /**
+   * Parses the given String into a Query instance.
+   */
+  def query(charset: Charset = UTF8, mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
```

Comment:
Ok, we can omit the charset. So we would have
```scala
private def query(mode: Uri.ParsingMode): Query = queryString match {
  case Some(q) ⇒ new UriParser(q, uriParsingMode = mode).parseQuery()
  case None    ⇒ Query.Empty
}
lazy val query = query(Uri.ParsingMode.Strict)
```
vs.
```scala
private def query(mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
  case Some(q) ⇒ new UriParser(q,  uriParsingMode = mode).parseQuery()
  case None    ⇒ Query.Empty
}
```
Does the lazy val really pull its weight here?